### ISSUE TITLE
If the Points are populated while the Plot has Visibility == Visibility.Hidden, it will not properly display them once it's visible again

### ISSUE THREAD

**Issue by Tyrrrz** [oxyplot/oxyplot#944]
*IF BUG, INCLUDE THIS PART:*

### Steps to reproduce

1. Have a Plot with visibility = hidden (collapsed works fine)
2. Populate the DataPoints either directly or through binding
3. Set the Visibility to Visible

Platform:
.NET version: 4.5

### Expected behaviour

The plot should show the points properly

### Actual behaviour

The plot displays empty canvas

### Observed with these conditions
1. Binding through ItemsSource binding on AreaSeries
2. DateTimeAxis on the bottom
3. LinearAxis on the left

Everything works fine, as long as Visibility != Hidden before the points are populated.

**Comment by objorke** [comment]
is this on WPF?

**Comment by Tyrrrz** [comment]
Yes this is on WPF

**Comment by marvel16** [comment]
Having the same issue!

**Comment by SaiPrashad** [comment]
Yes, having the same issue.... when the visibility of plot is hidden and making it visible and plotting the series at the same time then it does not appear the line, only the plot is visible. when i click the button 2nd time then the series is coming...
Please help...

**Comment by lemainan** [comment]
We've had the same problem in version 2.1. Dug into the OxyPlot WPF renderer and found that the implementation of OnRender in the WPF PlotView class will recreate the WPF elements during OnRender by calling RenderOverride. 

Given that OnRender happens as part of the Arrange step during WPF layout, this is a terrible choice, since this will invalidate measure & arrange of the child canvas. 
In our case, it even blocked rendering of the graph completely in some situations until you resized the PlotView . The reason is that the canvas can end up in a state where it needs to rerun arrange, but is not present in the arrange queue. Calling InvalidateArrange does nothing since the canvas is already marked as waiting for arrange.

We ended up completely overriding the Rendering context and drawing directly into a DrawingGroup that is used as backing store for the Canvas. Since all the position computation is done elsewhere anyway, there is no need to involve the WPF layout system.

As a side effect, this improves the performance significantly...
